metricflow/model/validations/reserved_keywords.py

```diff
@@ -0,0 +1,174 @@
+from typing import List
+from metricflow.instances import DataSourceElementReference
+
+
+from metricflow.model.objects.data_source import DataSource
+from metricflow.model.objects.user_configured_model import UserConfiguredModel
+from metricflow.model.validations.validator_helpers import (
+    DataSourceContext,
+    DataSourceElementContext,
+    DataSourceElementType,
+    FileContext,
+    ModelValidationRule,
+    ValidationError,
+    ValidationIssueType,
+    validate_safely,
+)
+
+# A non-exaustive tuple of reserved sql keywords
+RESERVED_KEYWORDS = (
```

Comment:
I forget exactly which source I pulled this list from. It was one of: [PSQL](https://www.postgresql.org/docs/current/sql-keywords-appendix.html), [IBM](https://www.ibm.com/docs/en/psfa/7.2.1?topic=keywords-sql-common-reserved-words), or [SQL Server](https://docs.microsoft.com/en-us/sql/t-sql/language-elements/reserved-keywords-transact-sql?view=sql-server-ver16), but I am unsure which. Additionally, I trimmed down the list to try and only have words that would be likely reserved in all engines. And I think I must have accidentally removed `USER` 🙃 I'm trying to _not_ make this rule engine dependent. We're also adding identifier and measure data warehouse validations in the next two weeks which will also catch these errors. The reason why we are adding this rule instead of just relying on data warehouse validations is because data warehouse validations are slow. So we're using this rule to catch the most common words to fail the model quickly.

Reply:
Alright so I updated the list by taking an intersection of PSQL, BigQuery, Redshift, and Snowflake keywords. It results in a smaller list. However, this also means we won't be blocking words when they aren't reserved for a specific engine (i.e. no false failures)